libs/datasets/timeseries.py

```diff
@@ -539,15 +550,23 @@ def timeseries_not_bucketed_wide_dates(self) -> pd.DataFrame:
 
     def _timeseries_latest_values(self) -> pd.DataFrame:
         """Returns the latest value for every region and metric, derived from timeseries."""
+
         if self.timeseries.columns.empty:
             return pd.DataFrame([], index=pd.Index([], name=CommonFields.LOCATION_ID))
         # timeseries is already sorted by DATE with the latest at the bottom.
         long = self.timeseries.stack().droplevel(CommonFields.DATE)
         # `long` has MultiIndex with LOCATION_ID and VARIABLE (added by stack). Keep only the last
         # row with each index to get the last value for each date.
         unduplicated_and_last_mask = ~long.index.duplicated(keep="last")
+
         return long.loc[unduplicated_and_last_mask, :].unstack()
 
+    @cached_property
+    def _timeseries_bucketed_latest_values(self) -> pd.DataFrame:
+        long_bucketed = self.timeseries_bucketed.stack().droplevel(CommonFields.DATE)
```

Comment:
How about adding a sort_index() between stack and droplevel to make sure it is sorted by date?